### ISSUE TITLE
Fractional scaling on Ubuntu breaks fullscreen

### ISSUE THREAD

**Issue by 7RST1** [ppy/osu-framework#4087]
Enabling fractional scaling on Pop OS 20.10 makes it so Osu doesn't fullscreen properly. The window stays at bottom left with rest of screen black. Choosing the resolution matching PC-screen, which is 1440p, in Window-mode makes it render at that resolution but still much smaller, and is far from covering the whole screen. It looks like when I start reaching 5K in resolution, it covers the whole screen. There is also some artifacting with Osu when moving it around in windowed mode, which might be related.
I have two screens set up;
1440p 144hz screen on which I am playing the game. Scaled 100%
4K 60hz secondary screen. Scaled 150%
I think this is an universal issue for everyone running fractional scaling on Ubuntu-based systems. I doubt my logs will come in handy as I think it's easy to reproduce.
Screenshot of fulscreen, and video of Osu in 1440p winodwed mode glitching out and too small are in details. The video is in the zip.

**osu!lazer version:** 
2020.1204.0

**Logs:**
[database.log](https://github.com/ppy/osu/files/5674326/database.log)
[network.log](https://github.com/ppy/osu/files/5674328/network.log)
[performance.log](https://github.com/ppy/osu/files/5674329/performance.log)
[runtime.log](https://github.com/ppy/osu/files/5674330/runtime.log)

<details>

![Screenshot of fullscreen with Osu too small](https://user-images.githubusercontent.com/16783057/101801473-aba17c00-3b0e-11eb-87c2-1a3a91e8501c.png)

[Zip with Osu in 1440p winodwed mode glitching out and too small](https://github.com/ppy/osu/files/5674534/osuGlitchingOut.zip)

</details>

**Comment by bdach** [comment]
Sure it's fractional scaling, rather than just any scaling at all?

**Comment by 7RST1** [comment]
Yes. The game fullscreens fine when scaling to 200% without fractional scaling.

**Comment by bdach** [comment]
Would be useful to know what in particular is used to achieve this fractional scaling (plain `xrandr`? `gnome-tweaks`?)

**Comment by 7RST1** [comment]
It has been built into Ubuntu since April, which makes it even bigger of an issue imo

**Comment by bdach** [comment]
Thanks for the screenshot, but as an aside, saying that it's "even bigger of an issue" is not really helpful in any capacity,

**Comment by 7RST1** [comment]
Just gave some "dramatic context", but yeah, there isn't really much more useful stuff I can provide. It should be pretty easy to reproduce though, so that people more familiar with the Osu lazer code and the underlying works of fractional scaling can have a shot at fixing it.

**Comment by 7RST1** [comment]
This is embarrasing; I posted the issue in the wrong repo. I have no idea how I managed to do that. I apologize!!

**Comment by bdach** [comment]
Um, no, it's in the correct place now, I transferred it here on purpose.

**Comment by 7RST1** [comment]
Oh XD I didn't notice. Is there any more context I can give?

**Comment by bdach** [comment]
I think the information provided here from your side should be enough to work with if you're not willing to explore yourself - just need to set up an environment with the fractional scaling option depicted as above available.

**Comment by faizalam9883** [comment]
Any update on this. I have the same problem. I discovered it on VLC. My screen stutters when in full screen and when moving the mouse pointer in full screen.

**Comment by bdach** [comment]
1. No.
2. What does VLC have to do with this?
3. "Screen stuttering" is a separate issue entirely.

**Comment by faizalam9883** [comment]
> 1. No.
> 2. What does VLC have to do with this?
> 3. "Screen stuttering" is a separate issue entirely.

I am saying that I discovered this issue on VLC. I was watching a movie and the screen started flickering. And when I exited the full screen it happened again.

**Comment by bdach** [comment]
This is not the issue tracker for VLC. This is the issue tracker for osu!lazer and osu!framework-powered projects. Please do not stray off topic.
